src/declarativewebcontainer.cpp

```diff
@@ -59,14 +60,26 @@ DeclarativeWebContainer::DeclarativeWebContainer(QQuickItem *parent)
     , m_completed(false)
     , m_initialized(false)
 {
-    m_webPages.reset(new WebPages(this));
     setFlag(QQuickItem::ItemHasContents, true);
+    setPrivateMode(m_settingManager->autostartPrivateBrowsing());
+
+    m_normalWebPages.reset(new WebPages(this));
+    m_privateWebPages.reset(new WebPages(this));
+
+    m_persistentTabModel.reset(new PersistentTabModel());
+    m_privateTabModel.reset(new PrivateTabModel());
```

Comment:
Here it makes sense to set instances' parent to `this`

m_persistentTabModel.reset(new PersistentTabModel(this));
m_privateTabModel.reset(new PrivateTabModel(this));